stats.go

```diff
@@ -107,3 +128,29 @@ func (s *statsExporter) submitMetric(v *view.View, row *view.Row, metricName str
 		return fmt.Errorf("aggregation %T is not supported", v.Aggregation)
 	}
 }
+
+func calculatePercentile(percentile float64, buckets []float64, countPerBucket []int64) float64 {
+	cumulativePerBucket := make([]int64, len(countPerBucket))
+	var sum int64
+	for n, count := range countPerBucket {
+		sum += count
+		cumulativePerBucket[n] = sum
+	}
+	atBin := int64(math.Floor(percentile * float64(sum)))
+
+	var previousCount int64
+	for n, count := range cumulativePerBucket {
+		if atBin >= previousCount && atBin <= count && n < len(buckets) {
+			return buckets[n]
+		}
+		previousCount = count
+	}
+	return buckets[len(buckets)-1]
+}
+
+func buildMetricNameForPercentile(percentile float64) string {
+	if percentile > 0.99 {
+		return fmt.Sprintf("%dpercentile", int64(percentile*1000+0.5))
```

Comment:
This might end up being called a lot, I'm not sure if it's worth it, but it might be interesting to experiment with a benchmark and possibly using https://golang.org/pkg/strconv/#FormatFloat. But I don't feel strongly about it. If you think this is enough, it's fine to leave it as it is for now. Just saying because `fmt.Sprintf` can be expensive.

Reply:
Used `strconv.Itoa` instead and changed the logic so we only compute percentile names when the exporter is created.